mypy/test/testcheck.py

```diff
@@ -222,6 +223,15 @@ def run_case_once(self, testcase: DataDrivenTestCase,
             if options.cache_dir != os.devnull:
                 self.verify_cache(module_data, res.errors, res.manager, res.graph)
 
+            if options.new_semantic_analyzer:
+                name = 'targets'
+                if incremental_step:
+                    name += str(incremental_step + 1)
+                expected = testcase.expected_fine_grained_targets.get(incremental_step + 1)
+                actual = res.manager.processed_targets
+                # Skip the initial builtin cycle.
+                actual = [t for t in actual if not any(t.startswith(mod) for mod in core_modules)]
+                assert_target_equivalence(name, expected, actual)
```

Comment:
It's pretty confusing that this does nothing if `expected` is None. Can you refactor this so that the `None` check happens in the caller instead?

Reply:
FWIW this matches the behaviour of a similar helper `assert_module_equivalence()` for `[rechecked]` and `[stale]`. I would rather either change them both, or leave them as is.

Reply:
Sure, it's fine to modify both (or neither).

Reply:
OK, I removed it from both and moved to the callers.